### PR TITLE
DW-5431 Add checking for tooling permissions in rds db

### PR DESCRIPTION
**Pull request by draagc-iw**

### CODE REVIEW

File: src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsClients.kt [hunk 1]
```diff
@@ -32,5 +34,6 @@ class AwsClients {
         iamClient = IamClient.builder().region(Region.AWS_GLOBAL).build()
         dynamoDbClient = DynamoDbClient.builder().region(configurationResolver.awsRegion).build()
         kmsClient = KmsClient.builder().region(configurationResolver.awsRegion).build()
+        rdsDataClient = RdsDataClient.builder().region(configurationResolver.awsRegion).build()
     }
 }
```
Comment (MattToczek) [inline src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsClients.kt:39]: Missing line

File: src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsCommunicator.kt [hunk 2]
```diff
@@ -540,4 +549,23 @@ class AwsCommunicator {
...
+        try{
+            val executeStatementRequest = ExecuteStatementRequest
+                .builder()
+                .secretArn(configurationResolver.getStringConfig(ConfigKey.RDS_CREDENTIALS_SECRET_ARN))
+                .resourceArn(configurationResolver.getStringConfig(ConfigKey.RDS_CLUSTER_ARN))
+                .database(configurationResolver.getStringConfig(ConfigKey.RDS_DATABASE_NAME))
+                .parameters(parameters)
+                .sql(sql)
+
+            return awsClients.rdsDataClient.executeStatement(executeStatementRequest.build())
```
Comment (MattToczek) [inline src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsCommunicator.kt:564]: Is there a reason that we are not closing the `builder` in the request?

File: src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsCommunicator.kt [hunk 3]
```diff
@@ -540,4 +549,23 @@ class AwsCommunicator {
...
+        } catch (e: SystemArgumentException){
```
Comment (MattToczek) [inline src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsCommunicator.kt:565]: [The Docs](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/rdsdata/RdsDataClient.html#executeStatement--) don't show this as an Exception thrown by this call - is this what has been thrown in the tests?

File: src/test/kotlin/uk/gov/dwp/dataworks/services/UserAuthorizationServiceTest.kt [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        )
+        val rdsDataResponse = ExecuteStatementResponse.builder()
+            .records(mutableListOf(record))
+            .build()
+
+        whenever(awsCommunicator.rdsExecuteStatement(any(), anyOrNull())).doReturn(rdsDataResponse)
+        assertFalse(userAuthorizationService.hasUserToolingPermission("testuser", ToolingPermission.FILE_TRANSFER_DOWNLOAD))
+    }
+
+}
```
Comment (MattToczek) [inline src/test/kotlin/uk/gov/dwp/dataworks/services/UserAuthorizationServiceTest.kt:136]: missing line

File: src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsCommunicator.kt (hunk 3 above)
Comment (draagc-iw) [reply src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsCommunicator.kt:565]: SystemArgumentException is thrown by the configuration resolver when a value is missing

File: src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsCommunicator.kt (hunk 2 above)
Comment (draagc-iw) [reply src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsCommunicator.kt:564]: There used to be a if statement that added the `.parameters()` to the request if not null, then realised I can just pass the possibly null value to the builder anyway, but forgot to change where the `build()` gets called. Doesn't really make any difference

**Review by MattToczek — DISMISSED** [review]

**Review by MattToczek — APPROVED** [review]

**Pull request merged**